notebook/_version.py

```diff
@@ -1,2 +1,28 @@
-version_info = (4, 1, 0, 'b1')
+"""
+store the current version info of the notebook.
+
+"""
+
+# Downstream maintainer, when running `python.setup.py jsversion`,
+# the version string is propagated to the JavaScript files,  do not forget to
+# patch the JavaScript files in `.postN` release done by distributions. 
+
+# Next beta/alpha/rc release: The version number for beta is X.Y.ZbN **without dots**. 
+
+version_info = (4, 1, '0b1')
 __version__ = '.'.join(map(str, version_info))
+
+import re
+
+pep440re = re.compile('^(\d+)\.(\d+)\.(\d+((a|b|rc)\d+)?)(\.post\d+)?(\.dev\d+)?$')
+
+
+
+def raise_on_bad_version(version):
```

Comment:
I understand where you're coming from, but my vote would be for this to be in the tests, and encourage people to run them as the very last step before uploading a release, rather than doing the check on every import.